modules/core/keeper/msg_server.go

```diff
@@ -741,7 +741,75 @@ func (k Keeper) ChannelUpgradeInit(goCtx context.Context, msg *channeltypes.MsgC
 
 // ChannelUpgradeTry defines a rpc handler method for MsgChannelUpgradeTry.
 func (k Keeper) ChannelUpgradeTry(goCtx context.Context, msg *channeltypes.MsgChannelUpgradeTry) (*channeltypes.MsgChannelUpgradeTryResponse, error) {
-	return nil, nil
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	module, chanCap, err := k.ChannelKeeper.LookupModuleByChannel(ctx, msg.PortId, msg.ChannelId)
+	if err != nil {
+		ctx.Logger().Error("channel upgrade try callback failed", "port-id", msg.PortId, "error", errorsmod.Wrap(err, "could not retrieve module from port-id"))
+		return nil, errorsmod.Wrap(err, "could not retrieve module from port-id")
+	}
+
+	cbs, ok := k.Router.GetRoute(module)
+	if !ok {
+		ctx.Logger().Error("channel upgrade try callback failed", "port-id", msg.PortId, "error", errorsmod.Wrapf(porttypes.ErrInvalidRoute, "route not found to module: %s", module))
+		return nil, errorsmod.Wrapf(porttypes.ErrInvalidRoute, "route not found to module: %s", module)
+	}
+
+	upgradeSequence, previousVersion, err := k.ChannelKeeper.ChanUpgradeTry(
+		ctx,
+		msg.PortId,
+		msg.ChannelId,
+		chanCap,
+		msg.CounterpartyChannel,
+		msg.CounterpartySequence,
+		msg.ProposedUpgradeChannel,
+		msg.TimeoutHeight,
+		msg.TimeoutTimestamp,
+		msg.ProofChannel,
+		msg.ProofUpgradeTimeout,
+		msg.ProofUpgradeSequence,
+		msg.ProofHeight,
+	)
+
+	if errorsmod.IsOf(err, channeltypes.ErrUpgradeAborted, channeltypes.ErrInvalidUpgradeSequence, channeltypes.ErrUpgradeTimeout) {
```

Comment:
Here we need to handle certain errors which signal that an error receipt has been written and must be committed. If we return an error to sdk baseapp then state changes are reverted

Reply:
One thought is to wrap any individual error that should be caught here in `ErrUpgradeAborted` and do a single check.